index.js

```diff
@@ -62,7 +63,8 @@ function fastifyWebsocket (fastify, opts, next) {
 
   function handleRouting (connection, request) {
     const response = new ServerResponse(request)
-    request[kWs] = connection
+    request[kWs] = WebSocket.createWebSocketStream(connection)
+    request[kWs].socket = connection
```

Comment:
Oh!! That's why my PR wasn't working 😆
Good catch 💪🏻

Reply:
Now I understand why you've reverted some of your code :).